src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -4096,6 +4099,23 @@ public IndexingStatsAggregator GetLatestIndexingStat()
 
         public abstract void SaveLastState();
 
+        
+        protected void HandleSourceDocumentIncludedInMapOutput()
+        {
+            if (_alreadyNotifiedAboutIncludingDocumentInOutput || SourceDocumentIncludedInOutput == false || PerformanceHintsConfig.AlertWhenSourceDocumentIncludedInOutput == false)
+                return;
+            
+            DocumentDatabase.NotificationCenter.Add(PerformanceHint.Create(
+                DocumentDatabase.Name,
+                $"Index '{Name}' is including the origin document in output.",
+                $"Please verify index definitions and consider a re-design of your entities or indexes for better indexing performance.",
+                PerformanceHintType.Indexing,
+                NotificationSeverity.Info,
```

Comment:
I'd consider it to be pretty important so maybe let's put it as a warning to grab user's attention more

Reply:
Also I wonder if we shouldn't pass and check `numberOfOutputs` here. If this is a fanout index then it's even more obvious mistake and it's more dangerous so we can include that info into the notification.

Reply:
Well, we're sending it only once per Index. So` numberOfOutputs` would be only valid for the first document.

Reply:
Yes, but still it's common that fanout indexes produce multiple entries on each document. Then we could underline that source document is outputted multiple times - which absolutely makes no sense